### PR TITLE
Add 'Typelevel Scala' dialects

### PR DESCRIPTION
**Pull request by xavier-fernandez**

### CODE REVIEW

File: scalameta/dialects/src/main/scala/scala/meta/dialects/package.scala [hunk 1]
```diff
@@ -128,19 +128,37 @@ package object dialects {
     allowSpliceUnderscores = true // SI-7715, only fixed in 2.11.0-M5
   )
 
+  implicit val TypelevelScala211 = Scala211.copy(
```
Comment (xeno-by) [inline scalameta/dialects/src/main/scala/scala/meta/dialects/package.scala]: How about we make this a little less verbose: `Typelevel211`?

File: scalameta/dialects/src/main/scala/scala/meta/dialects/package.scala [hunk 2]
```diff
@@ -128,19 +128,37 @@ package object dialects {
...
+    allowLiteralTypes = true
+  )
+
   implicit val Paradise211 = Scala211.copy(
     allowInlines = true
   )
 
+  implicit val ParadiseTypelevelScala211 = Paradise211.copy(
```
Comment (xeno-by) [inline scalameta/dialects/src/main/scala/scala/meta/dialects/package.scala]: I suggest we build this from `Typelevel211`, not `Paradise211`.

File: scalameta/dialects/src/main/scala/scala/meta/dialects/package.scala (hunk 1 above)
Comment (xavier-fernandez) [inline scalameta/dialects/src/main/scala/scala/meta/dialects/package.scala]: ok

**Comment by xavier-fernandez** [comment]
@xeno-by I did an ammend with your feedback comments fixed.

**Comment by xeno-by** [comment]
LGTM! Thank you.

**Comment by dwijnand** [comment]
That dialect encoding simplification paying off :-) Nice

**Pull request merged**